File: SegurosAPI/src/main/java/com/clinica/api/seguros_service/model/Usuario.java
package com.clinica.api.seguros_service.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "usuario")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Long id;

    @Column(nullable = false, length = 60)
    private String nombre;

    @Column(nullable = false, length = 100)
    private String apellido;

    @Column(name = "fecha_nacimiento", nullable = false)
    private LocalDateTime fechaNacimiento;

    @Column(nullable = false, unique = true, length = 100)
    private String correo;

    @Column(length = 12)
    private String telefono;

    @Column(nullable = false, length = 100)
    private String contrasena;

    @Column(name = "id_rol", nullable = false)
    private Long rolId;

    public Long getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public LocalDateTime getFechaNacimiento() {
        return fechaNacimiento;
    }

    public String getCorreo() {
        return correo;
    }

    public String getTelefono() {
        return telefono;
    }

    public String getContrasena() {
        return contrasena;
    }

    public Long getRolId() {
        return rolId;
    }
}

File: citas-service-nuevo/src/main/kotlin/citas_service_nuevo/model/Usuario.kt
package com.clinica.api.citas.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "usuario")
data class Usuario(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    var id: Long? = null,

    @Column(nullable = false)
    var nombre: String,

    @Column(nullable = false)
    var apellido: String,

    @Column(name = "fecha_nacimiento", nullable = false)
    var fechaNacimiento: LocalDateTime,

    @Column(nullable = false, unique = true)
    var correo: String,

    var telefono: String?,

    @Column(nullable = false)
    var contrasena: String,

    @Column(name = "id_rol", nullable = false)
    var idRol: Long
)